src/bidiMapper/domains/context/BrowsingContextProcessor.ts

```diff
@@ -335,89 +336,136 @@ export class BrowsingContextProcessor {
-        )
-        .catch((error) => this.#logger?.(LogType.debugError, error));
-      return;
-    }
-
     this.#logger?.(
       LogType.debugInfo,
       'AttachedToTarget event received:',
       params
     );
 
-    this.#setEventListeners(targetCdpClient);
+    switch (targetInfo.type) {
+      case 'page':
+      case 'iframe': {
+        if (targetInfo.targetId === this.#selfTargetId) {
+          break;
+        }
+
+        this.#setEventListeners(targetCdpClient);
+
+        const cdpTarget = CdpTarget.create(
+          targetInfo.targetId,
+          targetCdpClient,
+          this.#browserCdpClient,
+          sessionId,
+          this.#realmStorage,
+          this.#eventManager,
+          this.#preloadScriptStorage,
+          this.#networkStorage,
+          this.#acceptInsecureCerts
+        );
 
-    const maybeContext = this.#browsingContextStorage.findContext(
-      targetInfo.targetId
-    );
+        const maybeContext = this.#browsingContextStorage.findContext(
+          targetInfo.targetId
+        );
+        if (maybeContext) {
+          // OOPiF.
+          maybeContext.updateCdpTarget(cdpTarget);
+        } else {
+          // New context.
+          BrowsingContextImpl.create(
+            cdpTarget,
+            this.#realmStorage,
+            targetInfo.targetId,
+            null,
+            this.#eventManager,
+            this.#browsingContextStorage,
+            this.#logger
+          );
+        }
+        return;
+      }
+      case 'worker': {
+        this.#setEventListeners(targetCdpClient);
+
+        const cdpTarget = CdpTarget.create(
+          targetInfo.targetId,
+          targetCdpClient,
+          this.#browserCdpClient,
+          sessionId,
+          this.#realmStorage,
+          this.#eventManager,
+          this.#preloadScriptStorage,
+          this.#networkStorage,
+          this.#acceptInsecureCerts
+        );
 
-    const cdpTarget = CdpTarget.create(
-      targetInfo.targetId,
-      targetCdpClient,
-      this.#browserCdpClient,
-      sessionId,
-      this.#realmStorage,
-      this.#eventManager,
-      this.#preloadScriptStorage,
-      this.#networkStorage,
-      this.#acceptInsecureCerts
-    );
+        this.#createWorkerRealm(cdpTarget);
+        return;
+      }
+    }
 
-    if (maybeContext) {
-      // OOPiF.
-      maybeContext.updateCdpTarget(cdpTarget);
-    } else {
-      // New context.
-      BrowsingContextImpl.create(
-        cdpTarget,
+    // DevTools or some other not supported by BiDi target. Just release
+    // debugger and ignore them.
+    targetCdpClient
+      .sendCommand('Runtime.runIfWaitingForDebugger')
+      .then(() =>
+        parentSessionCdpClient.sendCommand('Target.detachFromTarget', params)
+      )
+      .catch((error) => this.#logger?.(LogType.debugError, error));
+  }
+
+  #workers = new Map<string, Realm>();
+  #createWorkerRealm(cdpTarget: CdpTarget) {
```

Comment:
This method does not create a worker realm, but rather adds listeners to create it after the corresponding event is received. Should it be called like `#createWorkerTarget` or `#processWorkerTarget` then?

Reply:
Creation is a subjective term. I prefer not to create an additional function simply for separating functionality since this requires additional lookup when reading the function.

In particular, splitting a function should be determined by two things:

 1. Number of times it is used is greater than 1.
 2. There is more than one functionality, and
 3. The function is long.

At least two of the factors above should be satisfied to justify splitting a function. In this case, 1 and 3 are not satisfied so I'd prefer not splitting it.

Reply:
Woops, read this wrong. Done.